detectron2/utils/visualizer.py

```diff
@@ -875,6 +875,7 @@ def draw_text(
             font_size = self._default_font_size
 
         # since the text background is dark, we don't want the text to be dark
+        color = [min(max(c, 0.0), 1.0) for c in color]
```

Comment:
As the documentation above says, `color` is not necessarily a list of float. Supported formats are https://matplotlib.org/stable/tutorials/colors/colors.html.

It's better to find where the out-of-bounds values are created and fix it there instead.